src/main/scala/org/constellation/storage/CheckpointService.scala

```diff
@@ -135,6 +138,66 @@ class CheckpointService[F[_]: Sync: LiftIO](
       (s: String) => ???
     )
 
+  def accept(checkpoint: FinishedCheckpoint)(implicit dao: DAO): F[Unit] = {
+
+    val peers = dao.readyPeers.map { ready =>
+      val filtered = ready.filter(t => checkpoint.facilitators.contains(t._1))
+      (if (filtered.isEmpty) ready else filtered)
+        .map(p => PeerApiClient(p._1, p._2.client))
+        .toList
+    }.unsafeRunSync()
+
+    (dao.nodeState, checkpoint.checkpointCacheData.checkpointBlock) match {
+      case (_, None) => Sync[F].raiseError(MissingCheckpointBlockException)
+      case (NodeState.Ready, Some(cb)) =>
+        for {
+          _ <- syncPending(pendingAcceptanceFromOthers, cb)
+          _ <- resolveMissingParents(cb, peers)
+          _ <- accept(checkpoint.checkpointCacheData)
+        } yield ()
+      case (NodeState.DownloadCompleteAwaitingFinalSync, Some(_)) =>
+        LiftIO[F].liftIO(dao.snapshotService.syncBufferAccept(checkpoint.checkpointCacheData))
+      case (_, Some(_)) => Sync[F].raiseError(PendingDownloadException(dao.id))
+    }
+  }
+
+  def resolveMissingParents(cb: CheckpointBlock, peers: List[PeerApiClient], depth: Int = 1)(
+    implicit dao: DAO
+  ): F[List[CheckpointCache]] = {
+
+    val checkError = if (depth >= maxDepth) {
+      Sync[F].raiseError(new Exception("Max depth reached when resolving data."))
+    } else Sync[F].unit
+
+    val resolveSoe = cb.parentSOEBaseHashes() match {
+      case List(_, _) => Sync[F].unit
+      case _          => LiftIO[F].liftIO(DataResolver.resolveSoe(cb.parentSOEHashes.toList, peers).void)
+    }
+
+    val resolveCheckpoint: F[List[CheckpointCache]] = Sync[F]
+      .delay(
+        cb.parentSOEBaseHashes().toList
+      )
+      .map(
+        parents =>
+          parents.traverse(h => LiftIO[F].liftIO(dao.checkpointService.contains(h)).map(exist => (h, exist)))
```

Comment:
why you're using `dao.checkpointService` even if this method is inside the `checkpointService`?